server/src/query/stream_schema_provider.rs

```diff
@@ -150,6 +150,10 @@ impl StandardTableProvider {
             }
         }
 
+        if manifest_files.is_empty() {
+            return Ok(None);
+        }
```

Comment:
We can return early, I think. If we move this to 130.

Reply:
is this addressed?

Reply:
There are calls to retain and truncate after 130. I think it is better to perform this check after all mutations to list have happened